### ISSUE TITLE
make flake8diff output differs from make lint-code

### ISSUE THREAD

**Issue by sam-falvo** [rackerlabs/otter#1346]
From https://travis-ci.org/rackerlabs/otter/builds/59311781:

2.77s$ make lint-code
git diff --patch --no-prefix HEAD^1 | flake8 --diff
pyflakes otter scripts autoscale_cloudcafe autoscale_cloudroast
otter/integration/tests/test_convergence.py:18: 'BreakLoopException' imported but unused
otter/integration/tests/test_convergence.py:28: 'retry_and_timeout' imported but unused
otter/integration/tests/test_convergence.py:30: 'TransientRetryError' imported but unused
otter/integration/tests/test_convergence.py:30: 'transient_errors_except' imported but unused
otter/integration/tests/test_convergence.py:30: 'repeating_interval' imported but unused
make: *** [lint-code] Error 1
The command "make lint-code" exited with 2.

From "make flake8diff":

(.ve)vagrant@vm:/opt/otter$ make flake8diff
git diff --patch --no-prefix 4a168c7a92dea8c194af78b3695d364ecf7b16df | flake8 --diff

Is there any way we can reconcile these differences?

**Comment by lvh** [comment]
You're running these in different environments, right? The top one is what's ran in Jenkins, the bottom one looks like it probably was run on your machine. lint-code runs flake8diff as a prerequisite, so it should be returning the same stuff (assuming it's done in the same environment).

**Comment by sam-falvo** [comment]
Correct; I'm running flake8diff inside of the dev-vm.

**Comment by lvh** [comment]
So, that code tries to diff against the thing that you're trying to merge into. On Travis, that's always a diff against HEAD^1; the HEAD commit is a fake merge commit to master (more or less). On your local machine, it's the branch-off point. I'm going to go ahead and guess that the branch-off point for the branch you're running this on isn't `master`?

**Comment by sam-falvo** [comment]
I'm not sure I understand what you're asking.  I created the branch via git checkout -b, but I'm reasonably sure my master isn't the latest (e.g., I didn't try git pull-ing first).  I should probably try that and see if I get the same results.  Will do that tomorrow.

**Comment by lvh** [comment]
It's not so much about how you created it; it's about where it branched off from (which could be master, or a different branch). It shouldn't matter much that it isn't the latest; that command will try and find the place where it branched off from (under the assumption that it will probably go back into that branch) and only show lint issues introduced since then. Are you trying to merge it into a different branch than what it branched off from?

**Comment by sam-falvo** [comment]
No, I believe I am definitely targeting master for this particular PR.
